lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go

```diff
@@ -689,9 +796,31 @@ func makeSubscriberState(subscriberID string, states state_types.StatesByID) *su
 	return sub.State
 }
 
+func concatenateSubscriberStates(s1 state_types.StatesByID, s2 state_types.StatesByID) state_types.StatesByID {
+	ret := state_types.StatesByID{}
```

Comment:
I don't think there's any need to create a new map. We can just merge s2 directly into s1

As an aside, a tiny nit: over time I've decided against using `ret` as a variable name. Cause when you read through code with `ret`, you have to then search up through the rest of the code to see what the variable semantically contains. So e.g. `merged := ...` is easier to quickly grok